sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -174,6 +175,289 @@ def create_schema(
             kind="DATABASE",
         )
 
+    def _insert_overwrite_by_condition(
+        self,
+        table_name: TableName,
+        source_queries: t.List[SourceQuery],
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        where: t.Optional[exp.Condition] = None,
+        insert_overwrite_strategy_override: t.Optional[InsertOverwriteStrategy] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        """
+        Implements the table or partition swap approach to insert-overwriting records.
+
+        Because this method executes multiple variants (full table replace, replace by time
+        range, replace by key, replace by partition), some upstream caller info is needed and
+        passed via kwargs.
+
+        Args:
+            table_name: Name of target table
+            source_queries: Source queries returning records to insert
+            columns_to_types: Column names and data types of target table
+            where: SQLGlot expression determining which target table rows should be overwritten
+            insert_overwrite_strategy_override: Not used by Clickhouse
+            kwargs:
+                dynamic_key: Key columns (replace by key only)
+                dynamic_key_exp: Expression to build key (replace by key only)
+                dynamic_key_unique: Whether more than one record can exist per key value (replace by key only)
+
+                is_inc_by_partition: Whether to overwrite partitions with only new records (incremental by partition only)
```

Comment:
Would a better name for this be a boolean `replace` ?

Reply:
Ahh I see what you're saying. To me, "replace" is more about the data (vs eg "merge" or "append").

Technically if you replace a partition with an exact copy of itself, it would be "replaced" but nothing has really changed, so i'd consider it a no-op rather than a replace.

Anyway, I think `keep_existing_partition_rows` is still a more descriptive name than `is_inc_by_partition`